File: 0695-max-area-of-island/0695-max-area-of-island.cpp
#include <vector>
class Solution {
public:
    int dfs(vector<vector<int>>& grid, int x, int y){
        if(x < 0 || y < 0 || x >= grid.size() || y >= grid[0].size() || grid[x][y] == 0){
            return 0;
        }
        grid[x][y] = 0;
        return 1 + dfs(grid, x-1, y) + dfs(grid, x, y-1) + dfs(grid, x+1, y) + dfs(grid, x, y+1);
        
    }
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int maxIslandArea = 0;
        int m = grid.size();
        int n = grid[0].size();

        for(int i=0; i<m; i++){
            for(int j=0; j<n; j++){
                if (grid[i][j] == 1){
                    maxIslandArea = max(maxIslandArea, dfs(grid, i, j));
                }
            }
        }
        return maxIslandArea;
    }
};

File: 0695-max-area-of-island/0695-max-area-of-island.py
class Solution:

    def dfs(self, grid, i, j):
        if(i<0 or j<0 or i>=len(grid) or j>=len(grid[0]) or grid[i][j] == 0):
            return 0
        grid[i][j] = 0
        return 1 + self.dfs(grid, i-1, j) + self.dfs(grid, i, j-1) + self.dfs(grid, i+1, j) + self.dfs(grid, i, j+1)

    def maxAreaOfIsland(self, grid: List[List[str]]) -> int:
        # islands = 0
        # area = 0
        maxArea = 0
        m = len(grid)
        n = len(grid[0])
                
        for i in range(m):
            for j in range(n):
                if(grid[i][j] == 1):
                    # islands += 1
                    maxArea = max(self.dfs(grid, i, j), maxArea)
                    
        return maxArea